libs/qCC_io/ShpFilter.cpp

```diff
@@ -651,6 +718,405 @@ static CC_FILE_ERROR LoadPolyline(QDataStream &shpStream,
+	if (!vertices->reserve(vertCount))
+	{
+		delete vertices;
+		return nullptr;
+	}
+	for (int32_t j = 0; j < vertCount; ++j)
+	{
+		vertices->addPoint(points[firstIndex + j]);
+	}
+	vertices->setEnabled(false);
+	return vertices;
+}
+
+//! Creates the ccMesh
+ccMesh *createMesh(
+		const std::vector<CCVector3> &points,
+		const std::vector<ScalarType> &scalarValues,
+		int32_t firstIndex,
+		int32_t lastIndex
+)
+{
+	int32_t vertCount = lastIndex - firstIndex + 1;
+	if (vertCount < 3)
+	{
+		return nullptr;
+	}
+
+	ccPointCloud *vertices = buildVertices(points, firstIndex, lastIndex);
+	if (!vertices)
+	{
+		return nullptr;
+	}
+
+	if (!scalarValues.empty())
+	{
+		bool areAllValuesOfPartsNans = true;
+		if (firstIndex < scalarValues.size() && lastIndex < scalarValues.size())
+		{
+			areAllValuesOfPartsNans = std::all_of(scalarValues.begin() + firstIndex,
+			                                      scalarValues.begin() + lastIndex,
+			                                      [](ScalarType val)
+			                                      { return std::isnan(val); });
+		}
+		if (!areAllValuesOfPartsNans)
+		{
+			auto *sf = new ccScalarField("Measures");
+			if (!sf->reserveSafe(vertCount))
+			{
+				ccLog::Warning(
+						QString("[SHP] Mesh: not enough memory to load scalar values!"));
+				sf->release();
+			} else
+			{
+				for (int32_t j = 0; j < vertCount; ++j)
+				{
+					sf->addElement(scalarValues[j + firstIndex]);
+				}
+				sf->computeMinAndMax();
+				int sfIdx = vertices->addScalarField(sf);
+				vertices->setCurrentDisplayedScalarField(sfIdx);
+				vertices->showSF(true);
+				vertices->enableScalarField();
+			}
+		}
+	}
+
+	return new ccMesh(vertices);
+}
+
+//! Builds the patches that where read from a MultiPatch record of a Shapefile
+//! patches are build as meshes, each part is its own mesh
+//! The only supported Patches/Part types are Triangle_Fan & Triangle_Strip
+CC_FILE_ERROR buildPatches(
+		ccHObject &container,
+		const std::vector<int32_t> &startIndexes,
+		const std::vector<int32_t> &partTypes,
+		const std::vector<CCVector3> &points,
+		const std::vector<ScalarType> &scalarValues
+)
+{
+	size_t numParts = startIndexes.size();
+	size_t numPoints = points.size();
+
+
+	for (int32_t i = 0; i < numParts; ++i)
+	{
+		if (!isValidEsriPartType(partTypes[i]))
+		{
+			ccLog::Warning("[SHP] Multipatch part %d has an invalid part type (%d)", i, partTypes[i]);
+			continue;
+		}
+		auto type = static_cast<ESRI_PART_TYPE>(partTypes[i]);
+
+		const int32_t &firstIndex = startIndexes[i];
+		const int32_t &lastIndex = static_cast<const int32_t &>((i + 1 < numParts ? startIndexes[i + 1] : numPoints) -
+		                                                        1);
+		const int32_t vertCount = lastIndex - firstIndex + 1;
+
+		switch (type)
+		{
+			case ESRI_PART_TYPE::TRIANGLE_STRIP:
+			{
+				ccMesh *mesh = createMesh(points, scalarValues, firstIndex, lastIndex);
+				for (int32_t j(2); j < vertCount; ++j)
```

Comment:
`int32_t j(2)` -> `int32_t j = 2`

I know it's the same thing, but the latter is the form we use everywhere.